### PR TITLE
Add support for old Mac OS X (native MIDI)

### PR DESCRIPTION
**Pull request by andyvand**
Add support for old Mac OS X (native MIDI)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -376,9 +376,9 @@ if(SDL_MIXER_X_STATIC AND NOT BUILD_AS_VB6_BINDING)
     if(NOT CMAKE_VERSION VERSION_LESS 3.0.2)
         set_target_properties(SDL2_mixer_ext_Static
             PROPERTIES
-            ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/$<$<CONFIG:Debug>:>"
```
Comment (Wohlstand) [inline CMakeLists.txt:379]: Don't remove these lines, they are workaround for MSBuild to don't put binaries into sub-directories and put them at the same place as everywhere.

File: CMakeLists.txt [hunk 2]
```diff
@@ -426,9 +426,9 @@ if(SDL_MIXER_X_SHARED AND NOT BUILD_AS_VB6_BINDING)
     if(NOT CMAKE_VERSION VERSION_LESS 3.0.2)
         set_target_properties(SDL2_mixer_ext
             PROPERTIES
-            ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib/$<$<CONFIG:Debug>:>"
```
Comment (Wohlstand) [inline CMakeLists.txt:429]: Same as above. Don't remove these lines.

File: CMakeLists.txt [hunk 3]
```diff
@@ -474,9 +474,9 @@ if(BUILD_AS_VB6_BINDING)
     if(NOT CMAKE_VERSION VERSION_LESS 3.0.2)
         set_target_properties(SDL2_mixer_ext_VB6
             PROPERTIES
-            ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/sdl-mixer-vb6/$<$<CONFIG:Debug>:>"
```
Comment (Wohlstand) [inline CMakeLists.txt:477]: And the same: don't remove these lines.

**Review by Wohlstand — CHANGES_REQUESTED** [review]
Looks fine, however, you shouldn't remove the noted expression as it's important to put binaries into the same directory as everywhere also (it's a workaround for the MSBuild to don't produce debug/release subdirectories and just put binaries at the usual place. It's important for external software that uses the library as an external project).

**Comment by Wohlstand** [comment]
Anyway, if you really want to support very old CMake, you better use the versioning to don't use expressions on CMake older than given version number, and use them everywhere also.

**Comment by andyvand** [comment]
> Anyway, if you really want to support very old CMake, you better use the versioning to don't use expressions on CMake older than given version number, and use them everywhere also.

I have readded /$<$<CONFIG:Debug>:>, you can merge the changes

**Pull request merged**